provider/resource_edge_deployment.go

```diff
@@ -0,0 +1,42 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceEdgeDeployment() *schema.Resource {
+	return &schema.Resource{
+		Create:   createOrUpdateEdgeDeployment,
+		Read:     readEdgeDeployment,
+		Update:   createOrUpdateEdgeDeployment,
+		Delete:   deleteEdgeDeployment,
+		Importer: &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			"site_short_name": {
+				Type:        schema.TypeString,
+				Description: "Site short name",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func createOrUpdateEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	pm := m.(providerMetadata)
+	sc := pm.Client
+
+	d.SetId(d.Get("site_short_name").(string))
+
+	return sc.CreateOrUpdateEdgeDeployment(pm.Corp, d.Get("site_short_name").(string))
+}
+
+func readEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	return nil
+}
+
+func deleteEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	pm := m.(providerMetadata)
```

Comment:
what happens when m is not a providerMetadata? Is that possible?

Reply:
See https://github.com/signalsciences/terraform-provider-sigsci/pull/112#discussion_r1107840500